packages/core/src/new-api/internal/reconciliation/reconcileFutureSpecificReconciliations.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  CallExecutionState,
+  ContractAtExecutionState,
+  DeploymentExecutionState,
+  ExecutionState,
+  ReadEventArgumentExecutionState,
+  SendDataExecutionState,
+  StaticCallExecutionState,
+} from "../../types/execution-state";
+import { Future, FutureType } from "../../types/module";
+
+import { reconcileArtifactContractAt } from "./futures/reconcileArtifactContractAt";
+import { reconcileArtifactContractDeployment } from "./futures/reconcileArtifactContractDeployment";
+import { reconcileArtifactLibraryDeployment } from "./futures/reconcileArtifactLibraryDeployment";
+import { reconcileNamedContractAt } from "./futures/reconcileNamedContractAt";
+import { reconcileNamedContractCall } from "./futures/reconcileNamedContractCall";
+import { reconcileNamedContractDeployment } from "./futures/reconcileNamedContractDeployment";
+import { reconcileNamedLibraryDeployment } from "./futures/reconcileNamedLibraryDeployment";
+import { reconcileNamedStaticCall } from "./futures/reconcileNamedStaticCall";
+import { reconcileReadEventArgument } from "./futures/reconcileReadEventArgument";
+import { reconcileSendData } from "./futures/reconcileSendData";
+import { ReconciliationContext, ReconciliationFutureResult } from "./types";
+
+export function reconcileFutureSpecificReconciliations(
+  future: Future,
+  executionState: ExecutionState,
+  context: ReconciliationContext
+): ReconciliationFutureResult {
+  switch (future.type) {
+    case FutureType.NAMED_CONTRACT_DEPLOYMENT:
+      return reconcileNamedContractDeployment(
+        future,
+        executionState as DeploymentExecutionState,
```

Comment:
Is this what you'd like to type more precisely?

Reply:
Yes, the types between future and execution state have a mirroring hierarchy. I would prefer to avoid the casting; however it is not a big risk given the tests.